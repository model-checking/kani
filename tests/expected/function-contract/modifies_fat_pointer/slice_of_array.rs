// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts

// Test that modifies a slice containing u32 size data

#[kani::modifies(&x[0..3])]
#[kani::ensures(|_| x[0..3].iter().map(|v| *v == 0).fold(true,|a,b|a&b))]
fn zero(x: &mut [u32; 6]) {
    x[0..3].fill(0)
}

#[kani::proof_for_contract(zero)]
fn harness() {
    let mut x = [kani::any(), kani::any(), kani::any(), kani::any(), kani::any(), kani::any()];
    zero(&mut x);
}
