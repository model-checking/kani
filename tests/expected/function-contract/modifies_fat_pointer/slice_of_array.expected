assertion\
- Status: SUCCESS\
- Description: "|_| x[0..3].iter().map(|v| *v == 0).fold(true,|a,b|a&b)"\

VERIFICATION:- SUCCESSFUL
